src/instruments/options/finite_difference_pricer.rs

```diff
@@ -0,0 +1,491 @@
+                tridiagonal_matrix[i][1] * theta[i] 
+                - tridiagonal_matrix[i - 1][2] * tridiagonal_matrix[i][0] * theta[i - 1]
+            )
+        }
+
+        let mut phi: Vec<f64> = Vec::new();
+        phi.push(1.0);
+        phi.push(tridiagonal_matrix[last][1]);
+
+        for i in 1..(tridiagonal_matrix.len() - 1) {
+            phi.push(
+                tridiagonal_matrix[last - i][1]
+                * phi[i] 
+                - tridiagonal_matrix[last - i][2] 
+                * tridiagonal_matrix[last + 1 - i][0] 
+                * phi[i-1]
+            )
+        }
+
+        phi.push(
+            tridiagonal_matrix[0][0]
+            * phi[last]
+            - tridiagonal_matrix[0][1] 
+            * tridiagonal_matrix[1][0] 
+            * phi[last-1]
+        );
+
+        let theta_n = theta.pop().unwrap();
+        phi.pop();
+        phi.reverse();
+
+        let mut value: f64;
+        let mut inverse_matrix: Vec<Vec<f64>> = Vec::new();
+        let mut matrix_row: Vec<f64> = Vec::new();
+
+        for i in 0..tridiagonal_matrix.len() {
+            for j in 0..tridiagonal_matrix.len() {
+                value = (-1.0_f64).powi((i+j) as i32);
+                
+                if i < j {
+                    for k in i..j {
+                        value *= tridiagonal_matrix[k].last().unwrap();
+                    }
+                    value *= theta[i] * phi[j] / theta_n;
+
+                } else if i == j {
+                    value *= theta[i] * phi[i] / theta_n
+                } else {
+                    for k in j..i {
+                        value *= tridiagonal_matrix[k+1].first().unwrap();
+                    }
+                    value *= theta[j] * phi[i] / theta_n
+                }
+                matrix_row.push(value);
+            }
+            
+            inverse_matrix.push(matrix_row.clone());
+            matrix_row.clear()
+        }
+
+        inverse_matrix
+        
+    }
+
+    fn sub_diagonal(&self, scaler: f64) -> Box<dyn Fn(f64) -> f64 + '_> {
+        let function = move |m: f64| scaler * ((self.volatility.powi(2) * m.powi(2)) - (self.risk_free_rate * m));
+        Box::new(function)
+    }
+
+    fn diagonal(&self, scaler: f64) -> Box<dyn Fn(f64) -> f64 + '_> {
+        let function = move |m: f64| 1.0 + scaler * ((self.volatility.powi(2) * m.powi(2)) + self.risk_free_rate);
+        Box::new(function)
+    }
+
+    fn super_diagonal(&self, scaler: f64) -> Box<dyn Fn(f64) -> f64 + '_> {
+        let function = move |m: f64| scaler * ((self.volatility.powi(2)) * m.powi(2) + (self.risk_free_rate * m));
+        Box::new(function)
+    }
+
+    fn price_steps(&self) -> u32 {
+        (self.initial_price / 100.00) as u32 * 100 + 100
+    }
+
+    fn delta_t(&self, time_steps: u32) -> f64 {
+        self.time_to_maturity as f64 / ((365 * time_steps) as f64)
+    }
+
+    fn time_steps(&self) -> u32 {
+        ((self.time_to_maturity / 365) + 1) as u32 * 1000
```

Comment:
Is 1000 a standard choice ?

Reply:
1000 was chosen since it is sufficiently large, although the choice is arbitrary

Reply:
It would be better if the user can choose the mesh parameters themselves. 
E.g. [QuantLib Python FD pricer](https://quantlib-python-docs.readthedocs.io/en/latest/pricing_engines.html#fdblackscholesvanillaengine)